experiments/misc/ilamb_conversion.jl

```diff
@@ -0,0 +1,389 @@
+    expected_dims = ["lon", "lat", "time"]
+    data = get_data(
+        clima_ds,
+        clima_short_name,
+        clima_units,
+        conversion_factor,
+        expected_dims,
+        FT,
+    )
+    NCDatasets.defVar(
+        ilamb_ds,
+        short_name,
+        data,
+        Tuple(expected_dims),
+        attrib = var_attribs,
+    )
+    return nothing
+end
+
+"""
+    get_data(
+        clima_ds,
+        clima_short_name,
+        clima_units,
+        conversion_factor,
+        expected_dims
+    )
+
+Get the data from `clima_ds` with the name `clima_short_name`.
+
+Additional preprocessing includes converting to units that ILAMB expects with
+`conversion_factor` and permuting the dimensions with `expected_dims`.
+"""
+function get_data(
+    clima_ds,
+    clima_short_name,
+    clima_units,
+    conversion_factor,
+    expected_dims,
+    FT,
+)
+    data = Array(clima_ds[clima_short_name])
+
+    # Special cases for tsoil and swc to extract top 10 cm (z >= -0.1)
+    if clima_short_name in ("tsoil", "swc")
+        z = Array(clima_ds["z"])
+        all_dimnames = NCDatasets.dimnames(clima_ds[clima_short_name])
+        top_layer_indices = findall(val -> val >= -0.1, z)
+        length(top_layer_indices) >= 1 ||
+            error("No valid depths above -0.1m found for $(clima_short_name)")
+        z_idx = last(top_layer_indices)
+
+        index_tuple =
+            (dimname == "z" ? z_idx : Colon() for dimname in all_dimnames)
+        data = data[index_tuple...]
+    end
+
+    # Permute dimensions to match ILAMB
+    dimnames = filter(
+        dimname -> dimname in expected_dims,
+        NCDatasets.dimnames(clima_ds[clima_short_name]),
+    )
+    perm = indexin(expected_dims, collect(dimnames))
+    clima_ds[clima_short_name].attrib["units"] != clima_units && error(
+        "Units are not the same for $clima_short_name. Update the conversion factor and units in the mapping and run this script again",
+    )
+    data .*= conversion_factor
+    data = permutedims(data, perm)
+    return FT.(data)
+end
+
+"""
+    function make_ilamb_file_name(
+        clima_short_name,
+        start_date::Dates.DateTime,
+        end_date::Dates.DateTime,
+    )
+
+Make the file name for the NetCDF file for use in the ILAMB leaderboard.
+"""
+function make_ilamb_file_name(
+    clima_short_name,
+    start_date::Dates.DateTime,
+    end_date::Dates.DateTime,
+)
+    ilamb_short_name = ILAMB_VARIABLES[clima_short_name].short_name
+
+    formatted_start_date = Dates.format(start_date, "YYYYmm")
+    formatted_end_date = Dates.format(end_date, "YYYYmm")
+    return "$(ilamb_short_name)_Lmon_CliMA_historical_r1i1p1f1_gn_$(formatted_start_date)-$(formatted_end_date).nc"
+end
+
+"""
+    find_short_name(filepath)
+
+Given a `filepath` to a NetCDF file produced by ClimaDiagnostics, return the
+short name of the variable stored in the NetCDF file.
+"""
+function find_short_name(filepath)
```

Comment:
this is fine for this PR, but it would be nice if eventually this function lives in ClimaDiagnostics